pkg/providers/pricing/pricing.go

```diff
@@ -225,85 +227,87 @@ func (p *DefaultProvider) UpdateOnDemandPricing(ctx context.Context) error {
+func (p *DefaultProvider) fetchOnDemandPricing(ctx context.Context, additionalFilters ...pricingtypes.Filter) (map[string]float64, error) {
 	prices := map[string]float64{}
-	filters := append([]*pricing.Filter{
+	filters := append([]pricingtypes.Filter{
 		{
 			Field: aws.String("regionCode"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String(p.region),
 		},
 		{
 			Field: aws.String("serviceCode"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String("AmazonEC2"),
 		},
 		{
 			Field: aws.String("preInstalledSw"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String("NA"),
 		},
 		{
 			Field: aws.String("operatingSystem"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String("Linux"),
 		},
 		{
 			Field: aws.String("capacitystatus"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String("Used"),
 		},
 		{
 			Field: aws.String("marketoption"),
-			Type:  aws.String("TERM_MATCH"),
+			Type:  "TERM_MATCH",
 			Value: aws.String("OnDemand"),
 		}},
 		additionalFilters...)
 
-	err := p.pricing.GetProductsPagesWithContext(
-		ctx,
-		&pricing.GetProductsInput{
-			Filters:     filters,
-			ServiceCode: aws.String("AmazonEC2"),
-		},
-		p.onDemandPage(ctx, prices),
-	)
-	if err != nil {
-		return nil, err
+	input := &pricing.GetProductsInput{
+		Filters:     filters,
+		ServiceCode: aws.String("AmazonEC2"),
+	}
+
+	paginator := pricing.NewGetProductsPaginator(p.pricing, input)
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+
+		if err != nil {
+			return nil, fmt.Errorf("getting pricing data, %w", err)
+		}
+		prices = lo.Assign(prices, p.onDemandPage(ctx, output))
 	}
 
 	return prices, nil
 }
 
-func (p *DefaultProvider) spotPage(ctx context.Context, prices map[string]map[string]float64) func(output *ec2.DescribeSpotPriceHistoryOutput, b bool) bool {
-	return func(output *ec2.DescribeSpotPriceHistoryOutput, b bool) bool {
-		for _, sph := range output.SpotPriceHistory {
-			spotPriceStr := aws.StringValue(sph.SpotPrice)
-			spotPrice, err := strconv.ParseFloat(spotPriceStr, 64)
-			// these errors shouldn't occur, but if pricing API does have an error, we ignore the record
-			if err != nil {
-				log.FromContext(ctx).V(1).Info(fmt.Sprintf("unable to parse price record %#v", sph))
-				continue
-			}
-			if sph.Timestamp == nil {
-				continue
-			}
-			instanceType := aws.StringValue(sph.InstanceType)
-			az := aws.StringValue(sph.AvailabilityZone)
-			_, ok := prices[instanceType]
-			if !ok {
-				prices[instanceType] = map[string]float64{}
-			}
-			prices[instanceType][az] = spotPrice
+func (p *DefaultProvider) spotPage(ctx context.Context, output *ec2.DescribeSpotPriceHistoryOutput, prices map[string]map[string]float64) map[string]map[string]float64 {
```

Comment:
```suggestion
func (p *DefaultProvider) spotPage(ctx context.Context, output *ec2.DescribeSpotPriceHistoryOutput, prices map[string]zonal) map[string]zonal {
```

Consider using the type since it makes it a little easier to read